### DISCUSSION TITLE
signTypedData MetaMask issue

### DISCUSSION THREAD

**Discussion by adjuric-symphony** [wevm/wagmi#2411] [Q&A]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Package Version
1.0.5

### Current Behavior

I'm trying to sign a message with useSignTypedData (signTypedDataAsync) and keep getting the error below even though I click 'Sign' on the MetaMask Signature request pop-up:

```
InternalRpcError: An internal error was received.

Details: MetaMask Typed Message Signature: User denied message signature.
Version: viem@0.3.33
    at withRetry.delay.count.count (buildRequest.ts:74:42)
    at async attemptRetry (withRetry.ts:34:22)
```

Any idea what could be the problem?

**Comment by mjRaza** [comment]
this is not a issue. user just rejected to generate a signature. this is the response what we get when user reject the signature.

**Reply by adjuric-symphony** [reply]
But I've clicked the 'Sign' button. More details: https://github.com/wagmi-dev/wagmi/issues/2412
